src/dusb/ti84series.js

```diff
@@ -83,6 +83,23 @@ module.exports = class Ti84series {
       flash: params.find(p => p.type == v.parameters.DUSB_PID_FREE_FLASH).value,
     };
   }
+  
+  async delete(name, type) {
+    await this._d.send({
+      type: v.virtualPacketTypes.DUSB_VPKT_MODIF_VAR,
+      data: [
+        0, name.length,
+        ...b.asciiToBytes(name),
+        ...b.constructParameters([{
+          type: v.attributes.DUSB_AID_VAR_TYPE2,
+          value: b.intToBytes(0xF0070000 + type, 4)
+        }]),
+        1,
+        0, 0, 0, 0, 0
+      ]
+    });
+    await this._d.expect(v.virtualPacketTypes.DUSB_VPKT_DATA_ACK);
```

Comment:
AFAICT, this, or expect() in device, does not handle error packets properly. See dusb_cmd_r_data_ack() in https://github.com/debrouxl/tilibs/blob/experimental2/libticalcs/trunk/src/dusb_cmd.cc .
Again, other parts of the code are already doing it wrong, but in order to enable proper long-term maintainability and growth of this code base, removing some of the excessive corner-cutting while adding other features is a good thing :)